src/core/surface/server.c

```diff
@@ -415,20 +419,29 @@ static void server_on_recv(void *ptr, int success) {
     case GRPC_STREAM_RECV_CLOSED:
       gpr_mu_lock(&chand->server->mu);
       if (calld->state == NOT_STARTED) {
+        grpc_iomgr_closure *kill_zombie_closure =
+            gpr_malloc(sizeof(grpc_iomgr_closure));
         calld->state = ZOMBIED;
-        grpc_iomgr_add_callback(kill_zombie, elem);
+        grpc_iomgr_managed_closure_init(kill_zombie_closure, kill_zombie, elem);
+        grpc_iomgr_add_callback(kill_zombie_closure);
       }
       gpr_mu_unlock(&chand->server->mu);
       break;
     case GRPC_STREAM_CLOSED:
       gpr_mu_lock(&chand->server->mu);
       if (calld->state == NOT_STARTED) {
+        grpc_iomgr_closure *kill_zombie_closure =
```

Comment:
We can guarantee that there's at most one kill_zombie outstanding at once - meaning we can probably squeeze this onto call_data.

It's probably going to be bad for memory usage, as this mechanism is used rarely. 
However: it lets us get rid of managed closures entirely, and when we get to memory usage optimization, maybe something presents itself as the obvious thing to do.